othello/moderator/moderator.py

```diff
@@ -0,0 +1,114 @@
+import enum
+
+from . import constants, utils
+
+
+class Player(enum.Enum):
+    BLACK = "x"
+    WHITE = "o"
+
+
+PLAYERS = {
+    0: Player.WHITE,
+    1: Player.BLACK,
+}
+
+INITIAL_BOARD = "...........................ox......xo..........................."
+
+
+class InvalidMoveError(RuntimeError):
+    def __init__(self, board, player, move):
+        self.code = utils.UserError.INVALID_MOVE.value[0]
+        self.message = utils.UserError.INVALID_MOVE.value[1].format(
+            move=move, player=PLAYERS[player].value, board=utils.binary_to_string(board)
+        )
+
+    def __str__(self):
+        return self.message
+
+
+class Moderator:
+    def __init__(self):
+        self.board = constants.INITIAL
+        self.game_over = False
+        self.current_player = constants.BLACK
+
+    def is_game_over(self):
+        return self.game_over
+
+    def get_board(self):
+        return utils.binary_to_string(self.board)
+
+    def outcome(self):
+        if self.is_game_over():
+            score = self.score()
+            return (
+                PLAYERS[constants.BLACK].value
+                if score > 0
+                else PLAYERS[constants.WHITE].value
+                if score < 0
+                else "T"
+            )
+        return False
+
+    def possible_moves(self):
+        discriminator = constants.FULL_BOARD ^ (
+            self.board[self.current_player] | self.board[1 ^ self.current_player]
+        )
+        moves = utils.bit_or(
+            utils.fill(self.board[self.current_player], self.board[1 ^ self.current_player], d)
+            & discriminator
+            for d in constants.MASKS
+        )
+        return moves
+
+    def make_move(self, move):
+        board = self.board.copy()
+        move = constants.MOVES[move]
+        board[self.current_player] |= move
+        opponent = 1 ^ self.current_player
+
+        for i in constants.MASKS:
+            c = utils.fill(move, board[opponent], i)
+            if c & board[self.current_player] != 0:
+                c = (
+                    (c & constants.MASKS[i * -1]) << i * -1
+                    if i < 0
+                    else (c & constants.MASKS[i * -1]) >> i
+                )
+                board[self.current_player] |= c
+                board[opponent] &= utils.bit_not(c)
+        self.board, self.current_player = board, opponent
+
+    def check_game_over(self):
+        current_moves = self.possible_moves()
+        self.current_player ^= 1
```

Comment:
You do this several times here and elsewhere; can you make it a separate method? Like `toggle_current_player()`?